openedx/core/djangoapps/catalog/utils.py

```diff
@@ -185,3 +185,39 @@ def get_course_runs():
         course_runs = get_edx_api_data(catalog_integration, 'course_runs', api=api, querystring=querystring)
 
     return course_runs
+
+
+def get_course_run_details(course_run_key, fields):
+    """
+    Retrieve information about the course run with the given id
+
+    Arguments:
+        course_run_key: key for the course_run about which we are retrieving information
+
+    Returns:
+        dict with language, start date, end date, and max_effort details about specified course run
+    """
+    catalog_integration = CatalogIntegration.current()
+    course_run_details = dict()
+    if catalog_integration.enabled:
```

Comment:
Nit: for long methods like this, it can sometimes be more readable to short circuit on a condition rather than wrapping everything in if/else. For example, you could do:
```
if not catalog_integration.enabled:
  msg = 'Unable to retrieve details about course_run {} because Catalog Integration is not enabled'.format(
    course_run_key
  )
  logger.error(msg)
  return course_run_details
try:
  user = catalog_integration.get_service_user()
  # other code
```

Reply:
Note: this is just my opinion, feel free to disregard.

Reply:
That definitely makes sense to me. I mainly just laid it out the way I did because it aligned with the existing util methods.